AdobeMediaGallery/Test/Unit/Model/Asset/Command/DeleteByIdTest.php

```diff
@@ -0,0 +1,177 @@
+                'mediaAssetFactory' =>  $this->mediaAssetFactory,
+                'logger' =>  $this->logger,
+            ]
+        );
+
+        $this->mediaAsset = $this->createMock(AssetInterface::class);
+        $this->adapter = $this->createMock(AdapterInterface::class);
+    }
+
+    /**
+     * Test delete media asset by id action
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testSuccessfulDeleteByIdExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $this->adapter->expects($this->at(0))
+            ->method('delete')
+            ->with($tableName, 'id = ' . $mediaAssetId);
+
+        $this->deleteMediaAssetById->execute($mediaAssetId);
+    }
+
+    /**
+     * Assume that delete action will thrown an Exception
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testExceptionOnDeleteExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $testErrorText = 'test error message';
+        $this->adapter->expects($this->at(0))
+            ->method('delete')
+            ->willThrowException(new \Exception($testErrorText));
+        $this->expectException(CouldNotDeleteException::class);
+        $testMessage = __(
+            'Could not delete media asset with id %id: %error',
+            [
+                'id' => $mediaAssetId,
+                'error' => $testErrorText,
+            ]
+        )->render();
+        $this->logger->expects($this->once())
+            ->method('critical')
+            ->with($testMessage)
+            ->willReturnSelf();
+        $this->expectExceptionMessage($testMessage);
```

Comment:
I wouldn't verify the test message. It will make the test fail even if small correction to exception message phrasing is applied